CMakeLists.txt

```diff
@@ -73,17 +75,82 @@ else() #If ROCBLAS not found
+
+find_package (hsa-runtime64)
+if(hsa-runtime64_FOUND)
+  message(STATUS "hsa-runtime64 found")
+  get_target_property(HSA_INC_DIR hsa-runtime64::hsa-runtime64 INTERFACE_INCLUDE_DIRECTORIES)
+  get_filename_component(HSA_PATH "${HSA_INC_DIR}/.." ABSOLUTE)
+  get_filename_component(HSA_LIB_DIR "${HSA_PATH}/lib" ABSOLUTE)
+  set(ROCR_INC_DIR ${HSA_INC_DIR})
+  set(ROCR_LIB_DIR ${HSA_LIB_DIR})
+else() #If HSA not found
+  message(FATAL_ERROR "hsa not found !!! Install hsa to proceed ...")
+endif(hsa-runtime64_FOUND)
+
+find_package (hsakmt)
+if(hsakmt_FOUND)
+  message(STATUS "hsakmt found")
+  get_target_property(ROCT_INC_DIR hsakmt::hsakmt INTERFACE_INCLUDE_DIRECTORIES)
+  get_filename_component(ROCT_LIB_DIR "${ROCT_INC_DIR}/../lib" ABSOLUTE)
+  message(STATUS "ROCT_INC_DIR: ${ROCT_INC_DIR}")
+  message(STATUS "ROCT_LIB_DIR: ${ROCT_LIB_DIR}")
+else() #If HSA not found
+  message(FATAL_ERROR "hsa not found !!! Install hsa to proceed ...")
+endif(hsakmt_FOUND)
+
+find_package (hip)
+if(hip_FOUND)
+  set(HIP_INC_DIR ${hip_INCLUDE_DIR})
+  get_filename_component(HIP_PATH "${hip_INCLUDE_DIR}/.." ABSOLUTE)
+  get_filename_component(HIP_LIB_DIR "${hip_INCLUDE_DIR}/../lib" ABSOLUTE)
+else() #If HIP not found
+  message(FATAL_ERROR "hip not found !!! Install hip to proceed ...")
+endif(hip_FOUND)
+
+find_package (rocm_smi)
+if(rocm_smi_FOUND)
+  message(STATUS "rocm_smi found")
+  set(ROCM_SMI_INC_DIR ${rocm_smi_INCLUDE_DIR} )
+  set(ROCM_SMI_LIB_DIR ${rocm_smi_LIB_DIR} )
+else() #If ROCM_SMI not found
+  message(FATAL_ERROR "rocm-smi not found !!! Install rocm-smi to proceed ...")
+endif(rocm_smi_FOUND)
+
+find_package (hiprand)
+if(hiprand_FOUND)
+  message(STATUS "hiprand found")
+  set(HIPRAND_INC_DIR ${hiprand_INCLUDE_DIR} )
+  get_filename_component(HIPRAND_LIB_DIR "${hiprand_INCLUDE_DIR}/../lib" ABSOLUTE)
+else() #If HIPRAND not found
+  message(FATAL_ERROR "hiprand not found !!! Install hiprand to proceed ...")
+endif(hiprand_FOUND)
+
+find_package (rocrand)
+if(rocrand_FOUND)
+  message(STATUS "rocrand found")
+  set(ROCRAND_INC_DIR ${rocrand_INCLUDE_DIR} )
+  get_filename_component(ROCRAND_LIB_DIR "${rocrand_INCLUDE_DIR}/../lib" ABSOLUTE)
+else() #If ROCRAND not found
+  message(FATAL_ERROR "rocrand not found !!! Install rocrand to proceed ...")
+endif(rocrand_FOUND)
+
 # Making ROCM_PATH, CMAKE_INSTALL_PREFIX, CPACK_PACKAGING_INSTALL_PREFIX as CACHE
 # variables since we will pass them as cmake params appropriately, and 
 # all find_packages relevant to this build will be in ROCM path hence appending it to CMAKE_PREFIX_PATH 
 set(ROCM_PATH "/opt/rocm" CACHE PATH "ROCM install path")
-set(CMAKE_INSTALL_PREFIX "/opt/rocm" CACHE PATH "CMAKE installation directory")
-set(CPACK_PACKAGING_INSTALL_PREFIX "/opt/rocm" CACHE PATH "Prefix used in built packages")
+set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+set(CPACK_PACKAGING_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")
```

Comment:
Not sure, if these changes are required, as this might have impact on how CI is using these to control packaging and install prefix separately.  As well the RPATH change may not be needed as the same can be passed as parameter in SPACK recipe

Reply:
Removed these changes and thunk dependency.